File: problems/70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int dp[46] = {0};
    int climbStairs(int n) {
        if(n<2) return 1;
        if(dp[n]) return dp[n];
        dp[n] = climbStairs(n-2) + climbStairs(n-1);
        return dp[n]; 
    }
};

File: problems/70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n < 3) return n;
        int a = 1;
        int b = 2;
        int ans = a + b;
        
        for(int i = 3; i < n; i++){
            a = b;
            b = ans;
            ans = a + b;
        }

        return ans;
    }
}
